python/ray/dag/dag_node.py

```diff
@@ -115,15 +130,31 @@ def execute(
                 - Serve handles for class nodes
                 - resolved values representing user input at runtime
         """
+        if compiled:
+            assert len(args) == 1, "Compiled DAGs support exactly one InputNode arg"
+            input_ref, output_channels = self.compiled()
+            input_ref.write(args[0])
+            return output_channels
 
         def executor(node):
             return node._execute_impl(*args, **kwargs)
 
-        result = self.apply_recursive(executor)
+        cache = {}
         if _ray_cache_refs:
+            cache = self.cache_from_last_execute
+        elif _ray_cache_actors:
+            for key, ref in self.cache_from_last_execute.items():
+                if isinstance(ref, ray.actor.ActorHandle):
+                    cache[key] = ref
+        result = self.apply_recursive(executor, cache=cache)
+        if _ray_cache_refs or _ray_cache_actors:
             self.cache_from_last_execute = executor.cache
         return result
 
+    def destroy_compiled_dag(self):
+        _, _, _, monitor = self.compiled()
+        monitor.destroy()
```

Comment:
Drop the fault tolerance/cancellation stuff from this PR?

Reply:
Oops, missed this...